injector/dns_disruption.go

```diff
@@ -229,8 +215,29 @@ func (i *DNSDisruptionInjector) Clean() error {
 		}
 	}
 
-	if err := i.config.Iptables.ClearAndDeleteChain("CHAOS-DNS"); err != nil {
-		return fmt.Errorf("unable to remove injected iptables chain: %w", err)
+	// check if there's still a jump rule to CHAOS-DNS chain existing
+	// that would block its deletion
+	rules, err := i.config.Iptables.ListRules("nat", "OUTPUT")
+	if err != nil {
+		return fmt.Errorf("error listing nat output rules: %w", err)
+	}
+
+	jumpRuleExists := false
+
+	for _, rule := range rules {
+		if strings.HasSuffix(rule, "-j CHAOS-DNS") {
```

Comment:
this assumes that `-j CHAOS-DNS` is the last parameter. Is it always the case?

we are using string `CHAOS-DNS` quite a bit, consider to extract a constant.

Reply:
nit - maybe change to strings.Contains to avoid relying on order of parameter.